traitsui/tests/editors/test_enum_editor.py

```diff
@@ -521,3 +426,40 @@ def test_list_evaluate_editor_text(self):
 
     def test_list_evaluate_editor_index(self):
         self.check_enum_index_update(get_evaluate_view("custom", mode="list"))
+
+    def check_list_enum_none_selected(self, view):
+        enum_edit = EnumModelWithNone()
+        view = View(
+            UItem(
+                "value",
+                editor=EnumEditor(
+                    values=["one", "two", "three", "four"],
+                    mode="list",
+                ),
+                style="custom",
+            ),
+            resizable=True,
+        )
+        tester = UITester()
+        with tester.create_ui(enum_edit, dict(view=view)) as ui:
+            self.assertEqual(enum_edit.value, None)
+            list_editor = tester.find_by_name(ui, "value")
+            displayed = list_editor.inspect(query.DisplayedSelectedText())
+            self.assertEqual(displayed, "")
+    
+    def test_list_enum_none_selected(self):
+        view = View(
+            UItem(
+                "value",
+                editor=EnumEditor(
+                    values=["one", "two", "three", "four"],
+                    mode="list",
+                ),
+                style="custom",
+            ),
+            resizable=True,
+        )
+        self.check_list_enum_none_selected(view)
+
+    def test_list_evaluate_none_selected(self):
+        self.check_list_enum_none_selected(get_evaluate_view("custom", mode="list"))
```

Comment:
I am very confused why these fail.  The similar test passes for the radio editor.  Also, on the displayed list, visually none of the entries are selected.  However, the first assertions fail (i.e. `enum_edit.value ='one'` to start, not `None`).  It may be the case that the List style does not allow None selection but Radio does, but that feels like strange/ incorrect behavior to me.

Reply:
I may need to just raise an error in the line you mentioned (i ended up returning "" as it seemed like it was possible to have nothing selected, but I may have been mistaken for the list editor style)

Reply:
On Qt, the failing line is: https://github.com/enthought/traitsui/blob/81902cc48f0af498d4f0f1a980dc8074da4d8265/traitsui/qt4/enum_editor.py#L428

Reply:
I don't believe this is failing because of anything UI Tester related.  It does seem like a bug in traitsui to me though.  It would be perfectly fine to not allow none of the enum options to be selected, but the inconsistency across styles of editors seems incorrect.  The trait being displayed has a default value of None, which is allowed, it seems strange for the editor to override this.

Reply:
Use of a format func was already tested separately, so I just left it as None explicitly.  Thus, it gets displayed as an option with the text 'None'.